### ISSUE TITLE
Report bugs to World4You dns api

### ISSUE THREAD

**Issue by NerLOR** [acmesh-official/acme.sh#3269]
This is the place to report bugs in the World4You DNS API.

If you experience a bug, please report it in this issue.

Thanks!

**Comment by thefox69** [comment]
get an error "Unable to parse paketnr".
used command: ./acme.sh --issue --dns dns_world4you -d example.com -d www.example.com
Tried it with different World4you domain/accounts

Logging in  is successfull, but paketno. cannot be read
Log extract:
[So Dez 20 19:32:25 CET 2020] Successfully logged in
[So Dez 20 19:32:25 CET 2020] GET
[So Dez 20 19:32:25 CET 2020] url='https://my.world4you.com/en/dashboard/paketuebersicht'
[So Dez 20 19:32:25 CET 2020] timeout=
[So Dez 20 19:32:25 CET 2020] _CURL='curl --silent --dump-header /root/.acme.sh/http.header  -L  -g '
[So Dez 20 19:32:26 CET 2020] ret='0'
[So Dez 20 19:32:26 CET 2020] **Unable to parse paketnr**

Thanks!

**Comment by NerLOR** [comment]
Bug already fixed. If you need the solution immediatly, use https://github.com/NerLOR/acme.sh 
I also made a pull request, so the fix should be soon available in the official repo.

**Comment by thefox69** [comment]
Thanks for fast response, just copied the dns_world4you.sh now i get an different error:

[So Dez 20 20:25:04 CET 2020] Using world4you to add record
[So Dez 20 20:25:04 CET 2020] Logging in...
[So Dez 20 20:25:06 CET 2020] Successfully logged in
[So Dez 20 20:25:08 CET 2020] Adding record...
[So Dez 20 20:25:10 CET 2020] Unable to add record: <!DOCTYPE html>     <html lang="en">        <head>      <meta charset="utf-8">          <meta http-equiv="Content-Type" content="text/html; charset=utf-8">            <meta name="viewport" content="width=device-width, initial-scale=1, shrink-to-fit=no">          <meta name="title" content="Der Kundenbereich von World4You">          <meta name="description" content="Im World4You Kundenbereich finden Sie alle ✔ wichtigen Informationen rund um Ihr Kundenkonto oder Details zu gebuchten Paketen. Klicken!">

**Comment by NerLOR** [comment]
Can you send me the full "HTML-Document"? (If you wish, you can obfuscate any sensitive information)

**Comment by thefox69** [comment]
Attached a copy of the html console output, so maybe not well formated
[html-reccorded.zip](https://github.com/acmesh-official/acme.sh/files/5721446/html-reccorded.zip)

**Comment by NerLOR** [comment]
Thank you. I tried to figure out, what went wrong, but I found nothing yet. For me the script works fine after the last fix and the GitHub tests didn't fail until now either. Your html-document is a ... strange response from world4you... (did you start multiple acme.sh/world4you instances at once or tried to log in in the web interface during the script ran?) 

I'll try to look into it in the next days.

**Comment by FloJ12** [comment]
[Thu Apr 28 07:57:13 CEST 2022] _CURL='curl --silent --dump-header /root/.acme.sh/http.header  --trace-ascii /tmp/tmp.XBzZd9M2i2  -g '
[Thu Apr 28 07:57:13 CEST 2022] _ret='0'
[Thu Apr 28 07:57:13 CEST 2022] GET
[Thu Apr 28 07:57:13 CEST 2022] url='https://my.world4you.com/en/6/dns'
[Thu Apr 28 07:57:13 CEST 2022] timeout=
[Thu Apr 28 07:57:13 CEST 2022] _CURL='curl --silent --dump-header /root/.acme.sh/http.header  -L  --trace-ascii /tmp/tmp.5iS0nWoXUO  -g '
[Thu Apr 28 07:57:14 CEST 2022] ret='0'
[Thu Apr 28 07:57:22 CEST 2022] Unable to add record: Unknown error
[Thu Apr 28 07:57:22 CEST 2022] Error add txt for domain:_acme-challenge.*************
[Thu Apr 28 07:57:22 CEST 2022] _on_issue_err
[Thu Apr 28 07:57:22 CEST 2022] Please add '--debug' or '--log' to check more details.
[Thu Apr 28 07:57:22 CEST 2022] See: https://github.com/acmesh-official/acme.sh/wiki/How-to-debug-acme.sh

Hello, I get following error when trying to renew a certificate with world4you dns challenge

**Comment by NerLOR** [comment]
Hi. This is strange... for me everything works just fine.

Are you sure, you work with the latest version? Is it possible, that you have 2FA enabled for your World4You account?
It seems, that the URL https://my.world4you.com/en/6/dns should contain more than just only this '6'...

Can you provide more information for this issue?

**Comment by FloJ12** [comment]
2FA is not enabled. 
It seems to me, that the txt entries are set, I can see them on the web portal, but it fails nevertheless

**Comment by NerLOR** [comment]
Is it possible, that you run multiple instances of this api at once or log in to World4You during execution? World4You only allowes one active session, all previous session will be automatically logged out. Perhaps something like this happens?

**Comment by FloJ12** [comment]
no, I have created a certificate with acme on Jan 13, 2022, which I now wanted to renew. So i run acme.sh --renew -d *** --force
I am not logged in anywhere

**Comment by FloJ12** [comment]
could it be that the issue is that the certificate has two domains in it? I validate srv1.***.at and srv2.***.at

**Comment by NerLOR** [comment]
Gernerally multi-domain certificates work. I use 'example.com' and '*.example.com' in one certificate. Acme simply executes the dns api for each domain, one after the other (as fas as I know).

**Comment by FloJ12** [comment]
okay. I also deactivated dnssec, but it didnt help.
what else could I try?
Issuing a new certificate fails with the same error

**Comment by NerLOR** [comment]
I found a small error in my code and fixed it: https://github.com/NerLOR/acme.sh/commit/12d9acd4b0b8d9ad82954632c07ecf37e736d6db

You can apply this change manually and should see some html files created for debugging. These might help you

**Comment by FloJ12** [comment]
Error-01.html: 
<!DOCTYPE html>
<html>
    <head>
        <meta charset="UTF-8" />
        <meta http-equiv="refresh" content="0;url='/en/paketauswahl/dns'" />

        <title>Redirecting to /en/paketauswahl/dns</title>
    </head>
    <body>
        Redirecting to <a href="/en/paketauswahl/dns">/en/paketauswahl/dns</a>.
    </body>
</html>

Error-02 is the DNS Entries area for me as customer. I am not sure if it helps you to find the error though and I don't want to scan through all the possible sensible data.

Is this expected behavior?

**Comment by NerLOR** [comment]
Yes, this is expected behavior. Error-02 may give you some insight on what went wrong. Try to search for some message box or something...

**Comment by NerLOR** [comment]
Hi! I updated the code again: https://github.com/NerLOR/acme.sh/commit/f6b2b4ab83441f75a108285d590a2f7da42a98c8

For me it now says: `Unable to add record: my.world4you.com: The chosen action has already been carried out too often today.`

So I will have to wait until tomorrow for further testing. But you might be able to get a more descriptive error message with the new fixes :)

**Comment by FloJ12** [comment]
[Thu Apr 28 19:52:18 CEST 2022] _CURL='curl --silent --dump-header /root/.acme.sh/http.header  -L  -g '
[Thu Apr 28 19:52:19 CEST 2022] ret='0'
[Thu Apr 28 19:52:19 CEST 2022] Unable to add record: my.world4you.com:
[Thu Apr 28 19:52:19 CEST 2022] Error add txt for domain:_acme-challenge.****
[Thu Apr 28 19:52:19 CEST 2022] _on_issue_err
[Thu Apr 28 19:52:19 CEST 2022] Please add '--debug' or '--log' to check more details.
[Thu Apr 28 19:52:19 CEST 2022] See: https://github.com/acmesh-official/acme.sh/wiki/How-to-debug-acme.sh
[Thu Apr 28 19:52:19 CEST 2022] url='https://acme-v02.api.letsencrypt.org/acme/chall-v3/103051323957/gR-7AQ'
[Thu Apr 28 19:52:19 CEST 2022] payload='{}'
[Thu Apr 28 19:52:20 CEST 2022] POST
[Thu Apr 28 19:52:20 CEST 2022] _post_url='https://acme-v02.api.letsencrypt.org/acme/chall-v3/103051323957/gR-7AQ'
[Thu Apr 28 19:52:20 CEST 2022] _CURL='curl --silent --dump-header /root/.acme.sh/http.header  -L  -g '
[Thu Apr 28 19:52:20 CEST 2022] _ret='0'
[Thu Apr 28 19:52:20 CEST 2022] code='200'
[Thu Apr 28 19:52:20 CEST 2022] socat doesn't exist.
[Thu Apr 28 19:52:21 CEST 2022] Diagnosis versions:
openssl:openssl
OpenSSL 1.0.2k-fips  26 Jan 2017
apache:
apache doesn't exist.
nginx:
nginx doesn't exist.
socat:
[Thu Apr 28 19:52:21 CEST 2022] pid
[Thu Apr 28 19:52:21 CEST 2022] No need to restore nginx, skip.
[Thu Apr 28 19:52:21 CEST 2022] _clearupdns
[Thu Apr 28 19:52:21 CEST 2022] dns_entries
[Thu Apr 28 19:52:21 CEST 2022] skip dns.

I am sorry, no new debug error messages

**Comment by NerLOR** [comment]
Can you verify, that the number in `/en/<NR>/dns` actually is your real "Paketnummer"? (For me it's in the format of `3XXXXXXX`)

**Comment by FloJ12** [comment]
yes, you are right. If I go in the dns panel on w4y, there it says 3XXXXXX
if I enter /en/6/dns when logged in, I am redirected to that number

**Comment by FloJ12** [comment]
could it be that there is a problem with the new webhosting paket which is named Webhosting Complete 6 ? 
on the top of the panel it says Webhosting Complete 6, 3XXXXXXX

**Comment by NerLOR** [comment]
I tried to fix `_parse_paket_nr`: https://github.com/NerLOR/acme.sh/commit/ed0cbc5066772ccb000f6712a344bf5d81dea2f5

please check, if this works for you.

**Comment by FloJ12** [comment]
It worked brilliantly!
Thank you very much

**Comment by NerLOR** [comment]
Great :D

**Comment by virus2500** [comment]
Hi, adding an challenge record works fine but removing it afterwards isn't.

```
[Sun Nov 24 00:38:44 CET 2024] Using world4you to remove record
[Sun Nov 24 00:38:44 CET 2024] Logging in...
[Sun Nov 24 00:38:46 CET 2024] Successfully logged in
[Sun Nov 24 00:38:48 CET 2024] Removing record...
[Sun Nov 24 00:38:49 CET 2024] Unable to remove record: my.world4you.com: An error has occurred.
__TITLE__
[Sun Nov 24 00:38:49 CET 2024] Error removing txt for domain: _acme-challenge.wildcardtest.mydomain.at
```

I guess the website code changed causing this issue?

From what i understand this script sends 

```
DeleteDnsRecordForm[recordId]=$recordid&
DeleteDnsRecordForm[uniqueFormIdDP]=$formiddp&DeleteDnsRecordForm[_token]=$form_token"
```

but the hidden fields in the modal to remove an entry are

```
<input type="hidden" id="DeleteDnsRecordForm_id" name="DeleteDnsRecordForm[id]" required="required" value="Q05BTUU6dGVzdC5teWRvbWFpbi5hdC46aG9tZS5teWRvbWFpbi5hdC4=">
<input type="hidden" id="DeleteDnsRecordForm_uniqueFormIdDP" name="DeleteDnsRecordForm[uniqueFormIdDP]" value="<remove>">
<input type="hidden" id="DeleteDnsRecordForm__token" name="DeleteDnsRecordForm[_token]" value="<removed>">
```

**Comment by virus2500** [comment]
Hi again, i fixed it for me. It was just changing DeleteDnsRecordForm[recordId] to DeleteDnsRecordForm[id].
Then removing the records worked as well.

**Comment by NerLOR** [comment]
Hi! Thank you for your detailed bug report. I reproduced the bug, verified your solution and will soon open a new pull request to fix the issue

**Comment by virus2500** [comment]
Awesome. Thank you :)

**Comment by beg9** [comment]
Hello,

first thanks for the implementation and I have tried it out and adding a challenge record works fine but removing not.
The Error Message “Unable to remove record: Unknown error” will be thrown.

I think the Problem is the used recordid (body="DeleteDnsRecordForm[id]=$recordid...).

Currently something like this is used:
<input type="hidden" id="DeleteDnsRecordForm_id" name="DeleteDnsRecordForm[id]" required="required" value="Q05BTUU6dGVzdC5teWRvbWFpbi5hdC46aG9tZS5teWRvbWFpbi5hdC4=">

But if I check the field on the Website a shorter id (10-digit number) is used, like:
<input type="hidden" id="DeleteDnsRecordForm_id" name="DeleteDnsRecordForm[id]" required="required" value="1342093720">

Could you please check that?

I tried on my own with the following Workaround and then it works. But to be honest, I don't really have any experience with this and I'm not sure if this is a reliable way to determine the relevant ID:

recordid=$(
     echo "$form" | tr '\n' ' ' \
       | grep -o 'data-id[[:space:]]*=[[:space:]]*"[^"]*"[[:space:]]*data-name[[:space:]]*=[[:space:]]*"[^"]*"[[:space:]]*class="delete-record"' \
       | grep "data-name=\"$fqdn\"" \
       | sed 's/^.*data-id[[:space:]]*=[[:space:]]*"\([^"]*\)".*$/\1/' \
       | head -n 1
   )

Thanks

**Comment by NerLOR** [comment]
Hi! Thank you for your message. I am encountering this error myself for a while now, but because everything kept working (except the "spam" of challenge records in the DNS) I ignored this issue until now. I fixed this error and opened a new PR: #6897 (currently waiting for all tests to pass), so this should be solved with the next release.

**Comment by beg9** [comment]
Thank you.

**Comment by CrazyChaoz** [comment]
where can i find those error log files ?

```
May 15 14:14:21 heimserver acme-deploy-start[2837007]: [Fri May 15 02:14:21 PM CEST 2026] Using world4you to remove record
May 15 14:14:22 heimserver acme-deploy-start[2837007]: [Fri May 15 02:14:22 PM CEST 2026] Logging in...
May 15 14:14:24 heimserver acme-deploy-start[2837007]: [Fri May 15 02:14:24 PM CEST 2026] Successfully logged in
May 15 14:14:26 heimserver acme-deploy-start[2837007]: [Fri May 15 02:14:26 PM CEST 2026] Removing record...
May 15 14:14:28 heimserver acme-deploy-start[2837007]: [Fri May 15 02:14:28 PM CEST 2026] Unable to remove record: Unknown error
May 15 14:14:28 heimserver acme-deploy-start[2837007]: [Fri May 15 02:14:28 PM CEST 2026] View error-01.html and error-02.html for debugging
```

**Comment by neilpang** [comment]
Closing this tracking issue for now to keep the issue list clean. It remains the place to report problems with this provider -- if you hit a bug, comment here and the issue will be reopened.
